### DISCUSSION TITLE
API documentation,  am I doing something wrong?

### DISCUSSION THREAD

**Discussion by jzandbergen** [netbox-community/netbox#7433] [Q&A] [answered]
Hi All,

We're currently running NetBox 2.11.12 and I was wondering if the following is intended behavior for the api.

I have 2 clusters defined, one name "oVirt" and the other "test". In both clusters I have a virtual machine named `testfromsalt1` . I was under the impression when reading the NetBox API that I could do:
```
https://netbox.[redacted]/api/virtualization/virtual-machines/?name=testfromsalt1&cluster=test
```
But the response is:
```json
{
  "cluster": [
    "Select a valid choice. That choice is not one of the available choices."
  ]
}
```
If I do the request:
```
https://netbox.[redacted]/api/virtualization/virtual-machines/?name=testfromsalt1&cluster="1"
```
Where `1` is the `id` of the cluster, I get the expected output.

So my question is: "Is it correct that the `cluster` parameter needs an `id` to search? If so, what is the significance of the `cluster_id` parameter?".

Thank you for your time.

**Comment by jeremystretch** [comment, marked as answer]
It looks like the `cluster` field was added to the filter set for the VirtualMachine model in error. Typically, you would use e.g. `site_id=1` or `site=foo` to filter by a related object, where the later specifies an object by its unique slug (human-friendly name). However, the Cluster model doesn't have a slug, so this doesn't apply.

**Reply by jzandbergen** [reply]
Thank you very much! That also explains why in another case I was wondering why `tenant` had to be supplied in lowercase. :+1:
